src/cbc_sdk/endpoint_standard/base.py

```diff
@@ -368,3 +384,180 @@ def _search(self, start=0, rows=0):
+        return self
+
+    def _submit(self):
+        if self._query_token:
+            raise ApiError("Query already submitted: token {0}".format(self._query_token))
+
+        args = self._get_query_parameters()
+
+        url = "/api/investigate/v2/orgs/{}/enriched_events/search_jobs".format(self._cb.credentials.org_key)
+        query_start = self._cb.post_object(url, body=args)
+        self._query_token = query_start.json().get("job_id")
+
+        self._timed_out = False
+        self._submit_time = time.time() * 1000
+
+    def _still_querying(self):
+        if not self._query_token:
+            self._submit()
+
+        status_url = "/api/investigate/v1/orgs/{}/enriched_events/search_jobs/{}".format(
+            self._cb.credentials.org_key,
+            self._query_token,
+        )
+        result = self._cb.get_object(status_url)
+        searchers_contacted = result.get("contacted", 0)
+        searchers_completed = result.get("completed", 0)
+        log.debug("contacted = {}, completed = {}".format(searchers_contacted, searchers_completed))
+        if searchers_contacted == 0:
+            return True
+        if searchers_completed < searchers_contacted:
+            if self._timeout != 0 and (time.time() * 1000) - self._submit_time > self._timeout:
+                self._timed_out = True
+                return False
+            return True
+
+        return False
+
+    def _count(self):
+        if self._count_valid:
+            return self._total_results
+
+        while self._still_querying():
+            time.sleep(.5)
+
+        if self._timed_out:
+            raise TimeoutError(message="user-specified timeout exceeded while waiting for results")
+
+        result_url = "/api/investigate/v2/orgs/{}/enriched_events/search_jobs/{}/results".format(
+            self._cb.credentials.org_key,
+            self._query_token,
+        )
+        result = self._cb.get_object(result_url)
+
+        self._total_results = result.get('num_available', 0)
+        self._count_valid = True
+
+        return self._total_results
+
+    def _search(self, start=0, rows=0):
+        if not self._query_token:
+            self._submit()
+
+        while self._still_querying():
+            time.sleep(.5)
+
+        if self._timed_out:
+            raise TimeoutError(message="user-specified timeout exceeded while waiting for results")
+
+        log.debug("Pulling results, timed_out={}".format(self._timed_out))
+
+        current = start
+        rows_fetched = 0
+        still_fetching = True
+        result_url_template = "/api/investigate/v2/orgs/{}/enriched_events/search_jobs/{}/results".format(
+            self._cb.credentials.org_key,
+            self._query_token
+        )
+        query_parameters = {}
+        while still_fetching:
+            result_url = '{}?start={}&rows={}'.format(
+                result_url_template,
+                current,
+                10  # Batch gets to reduce API calls
```

Comment:
Why not more? Larger batches are faster :)

Reply:
Actually, there is `self._batch_size = 100` so it could be used here instead of the hardcoded value

Reply:
Done, changed as requested.

Reply:
You can close this yourself ;)